### PR TITLE
MAINT: Turn old string formatting into f-strings (part 1)

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Do not modify `f2py` yet, it's perhaps the most complex part.

### CODE REVIEW

**Comment by seberg** [comment]
The patch was created with an automated tool and largely manually verified?

File: numpy/core/records.py [hunk 1]
```diff
@@ -548,7 +548,7 @@ def __repr__(self):
                 self, separator=', ', prefix=prefix, suffix=',')
         else:
             # show zero-length shape unless it is (0,)
-            lst = "[], shape=%s" % (repr(self.shape),)
+            lst = f"[], shape={repr(self.shape)}"
```
Comment (seberg) [inline numpy/core/records.py:551]: I guess the automated tool isn't able to do the `!r` here?

**Comment by DimitriPapadopoulos** [comment]
This kind of change results in many conflicts to solve, both for me during the review and for other committers after the merge. Perhaps it's best to apply changes gradually, in small batches. This part 1.

**Comment by mattip** [comment]
I admit I am not a big fan of this kind of PR: it seems like churn I would like to avoid.
In any case, do not modify distutils, it is deprecated.

**Comment by DimitriPapadopoulos** [comment]
My idea was to clean up existing code base, so that _pyupgrade_ or _ruff_ can be added to CI jobs later on.

**Comment by seberg** [comment]
I want to be positive about this type of change in general, but I have to admit that we churning the repo a lot currently, and adding this on top feels like playing with fire especially at this time.

**Comment by DimitriPapadopoulos** [comment]
I understand. Is there a better time to push such changes? What if they were limited to parts of the code base (which parts?) that don't change that much?

**Comment by DimitriPapadopoulos** [comment]
I think I'll close this PR, in favour or smaller PR that applies ruff/pyupgrade `UP` rules that don't change the code that much. We will have to disable the ruff/pyupgrade rules that suggest changing string interpolation based on `%` and `format` to f-strings, at least in the foreseeable future.

**Pull request closed without merging**